src/test/enhetstester/kotlin/no/nav/familie/ks/sak/barnehagelister/BarnehagebarnServiceTest.kt

```diff
@@ -0,0 +1,411 @@
+                kunLøpendeFagsak = true,
+                kommuneNavn = null,
+            )
+
+        val infotrygdIdentListe = listOf("infotrygdIdent1", "infotrygdIdent2")
+        every { mockInfotrygdReplikaClient.hentAlleBarnasIdenterForLøpendeFagsaker() } returns infotrygdIdentListe
+
+        every {
+            mockBarnehagebarnRepository.findBarnehagebarnByIdentInfotrygd("eksisterendeIdent", infotrygdIdentListe, any())
+        } returns PageImpl(listOf(mocketBarnehagebarnInfotrygdDtoInterface))
+
+        // Act
+        val hentetMocketPageBarnehagebarnDtoInterface = barnehagebarnService.hentBarnehagebarnInfotrygd(barnehagebarnRequestParamsMedIdent)
+
+        // Assert
+        verify(exactly = 1) { mockBarnehagebarnRepository.findBarnehagebarnByIdentInfotrygd("eksisterendeIdent", infotrygdIdentListe, any()) }
+        verify(exactly = 1) { mockInfotrygdReplikaClient.hentAlleBarnasIdenterForLøpendeFagsaker() }
+
+        assertThat(hentetMocketPageBarnehagebarnDtoInterface.content.size).isEqualTo(1)
+        val barnehagebarnInfotrygdDto = hentetMocketPageBarnehagebarnDtoInterface.content.single()
+
+        assertThat(barnehagebarnInfotrygdDto.ident).isEqualTo("ident1")
+        assertThat(barnehagebarnInfotrygdDto.antallTimerIBarnehage).isEqualTo(40.0)
+        assertThat(barnehagebarnInfotrygdDto.harFagsak).isTrue()
+    }
+
+    @Test
+    fun `hentBarnehagebarnInfotrygd skal hente barn fra infotrygd uavhengig av fagsak fra kommunenavn dersom det sendes inn som parameter`() {
+        // Arrange
+        val mocketBarnehagebarnInfotrygdDtoInterface =
+            opprettMocketBarnehagebarnInfotrygdDtoInterface(
+                "ident1",
+                "endringsType",
+                "Oslo",
+                "0000",
+            )
+        val barnehagebarnRequestParamsMedIdent =
+            BarnehagebarnRequestParams(
+                ident = null,
+                kunLøpendeFagsak = false,
+                kommuneNavn = "eksisterendeKommune",
+            )
+
+        val infotrygdIdentListe = listOf("infotrygdIdent1", "infotrygdIdent2")
+        every { mockInfotrygdReplikaClient.hentAlleBarnasIdenterForLøpendeFagsaker() } returns infotrygdIdentListe
+
+        every {
+            mockBarnehagebarnRepository.findBarnehagebarnByKommuneNavnInfotrygdUavhengigAvFagsak("eksisterendeKommune", any())
+        } returns PageImpl(listOf(mocketBarnehagebarnInfotrygdDtoInterface))
+
+        // Act
+        val hentetMocketPageBarnehagebarnDtoInterface = barnehagebarnService.hentBarnehagebarnInfotrygd(barnehagebarnRequestParamsMedIdent)
+
+        // Assert
+        verify(exactly = 1) { mockBarnehagebarnRepository.findBarnehagebarnByKommuneNavnInfotrygdUavhengigAvFagsak("eksisterendeKommune", any()) }
+        verify(exactly = 1) { mockInfotrygdReplikaClient.hentAlleBarnasIdenterForLøpendeFagsaker() }
+
+        assertThat(hentetMocketPageBarnehagebarnDtoInterface.content.size).isEqualTo(1)
```

Comment:
Ser denne gjelder flere plasser også 

```suggestion
        assertThat(hentetMocketPageBarnehagebarnDtoInterface.content).hasSize(1)
```